autogen/code_utils.py

```diff
@@ -62,6 +62,10 @@ def extract_code(
           If there is no code block in the input text, the language would be "unknown".
           If there is code block but the language is not specified, the language would be "".
     """
+
+    # Some Local LLM models/servers output \r\n insteaf or just \n. Let's clean it up before continuing
+    text = re.sub(r"\r\n", "\n", text)
```

Comment:
Could we add a parameter or environment variable to control enable or disable this replacement?

Reply:
Was testing this morning using detect_single_line_code=True and it worked fine with and without \r\n since it's not looking for \n's.

Perhaps we should have it set to default since it seemingly works much better overall.

Reply:
Thanks @robzsaunders . My concern is that if in some case there is a line of code like `print("something\r\n")`, the replacement will modify the code itself while maybe the user wants to keep it.